Add import_aip management command

jrwdunham opened the pull request:
Example usage:

    $ make manage-ss ARG='import_aip /home/archivematica/new_uncompressed.tar.gz'
    docker-compose run \
    		--rm \
    		--entrypoint /src/storage_service/manage.py \
    			archivematica-storage-service \
    				import_aip /home/archivematica/new_uncompressed.tar.gz
    Starting compose_mysql_1 ... done
    Attempting to import the AIP at /home/archivematica/new_uncompressed.tar.gz.
    An AIP with UUID 0fdedcc1-599d-4f6e-b132-4b7bc0355b1f already exists in this Storage Service? If you want to import this AIP anyway (and destroy the existing one), then enter "y" or "yes": y
    Successfully imported AIP 0fdedcc1-599d-4f6e-b132-4b7bc0355b1f.

Connected to archivematica/Issues#24
Connected to https://github.com/archivematica/Issues/issues/449

Takes code from dev branch at https://github.com/artefactual/archivematica-devtools/tree/dev/issue-46-import_aip to create a Django management command in the SS that does the same thing.

May be related to https://github.com/artefactual/archivematica-storage-service/pull/143.

jrwdunham committed:
Add import_aip management command

sevein commented on storage_service/common/management/commands/import_aip.py:
```diff
@@ -0,0 +1,405 @@
+ANSI_OKGREEN = '\033[92m'
+ANSI_WARNING = '\033[93m'
+ANSI_FAIL = '\033[91m'
+ANSI_ENDC = '\033[0m'
```
You may want to revisit some of the stuff you're doing around logging. It'd be good to take a look at https://github.com/django/django/tree/stable/1.8.x/django/core/management and see how things are done in Django, e.g.: see `colors`, `DJANGO_COLORS`, exceptions like `CommandError`, etc...

sevein commented on storage_service/common/management/commands/import_aip.py:
```diff
@@ -0,0 +1,405 @@
+
+
+class ImportAIPException(Exception):
+    """An error occurred when attempting to import an AIP."""
```
Should this subclass `CommandError`?

> If this exception is raised during the execution of a management command, it will be caught and turned into a nicely-printed error message to the appropriate output stream (i.e., stderr); as a result, raising this exception (with a sensible description of the error) is the preferred way to indicate that something has gone wrong in the execution of a command.

sevein reviewed (commented):
I've tried this and it's working great. @hakamine is going to try this on the client side soon.

sevein committed:
Update bagits to v0.97 during reingest

This is to ensure that tag manifests are regenerated after payload
changes which is going to happen during full reingest. Without this
commit, bagit-python was not updating all tag manifests found in the
package.

sevein committed:
Fix flake8 warnings

hakamine committed:
black: format PR 396

hakamine reviewed (approved):
I rebased both AM and SS dev/issue-24-handle-old-aips branches on top of the current qa/1.x and deployed on a test server. I ran an import, a reindex, and reingest of AM68 without errors. I think the dev branches are ready to merge to AM a/1.x and SS qa/0.x (then a more extensive testing/refining can be done)

sevein commented:
:clap: :clap: :clap:
